vault/provider.go

```diff
@@ -68,6 +68,30 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("VAULT_CAPATH", ""),
 				Description: "Path to directory containing CA certificate files to validate the server's certificate.",
 			},
+			"auth_login": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Login to vault with an existing auth method using auth/<mount>/login",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"namespace": {
```

Comment:
In the Terraform Vault Provider presently, you can set the namespace _once_ per Terraform run. We've been thinking about making it more like this, where you could designate the namespace on the client per-resource or per-data source. 

I'm just curious if you have a use case where you'd have a different namespace for different login endpoints on the same run of Terraform? If so, would you be willing to share? I've not directly heard any of the use cases yet, though I know they exist, so I'm just wanting to learn so I can understand it more.

Reply:
Currently, the vault provider supports auth'ing to a namespace other than set in the provider if the token's namespace differs. (i.e. a token where s.<token-id>.<namespace-id> is provided)
Our current installation of vault uses Okta auth at the root namespace, but our teams are split into multiple namespaces. I make use of identity-groups and inheritance to translate an identity at the root namespace to policies in the team's namespace. 
Of course, this is only for user access, as an automated pipeline is hardly expected to use okta for logins.
However I expect a pattern to emerge: a team has a suite of sub-namespaces to control access to prod/nonprod secrets but controls programmatic access to these secrets by having a central approle or other auth method in the parent namespace. 
It is expected at this point that a terraform run would auth to 'ns1/' and perform actions in 'ns1/subns1'. 
Therefore, if we configure the provider like so:
```
provider vault {
  address = "something:8200"
  namespace = "ns1/subns1"
  auth_login {
    path = "auth/approle/login"
    namespace = "ns1"
    parameters = {...}
  }
}
```
One would expect it to get a token from the approle mount in 'ns1' and perform actions in 'ns1/subns1' assuming the policies it got from the initial login allow it to do so. 

Even if this pattern does not appear, it at least provides flexibility.

Reply:
By the way, I already designate namespaces on a per-resource, per-data-source basis by using aliased providers. 
e.g.
```
provider vault {
  alias = "ns1"
  namespace = "ns1"
}
provider vault {
  alias = "ns2"
  namespae = "ns2"
}
```
then:
```
resource vault_generic_secret {
  provider = "vault.ns1"
  ...
}
```
But the auth context can still be different for the sake of controlling the sources of identities.

Reply:
That is very helpful! Thank you!